lightstep/sdk/metric/config.go

```diff
@@ -0,0 +1,65 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric // import "github.com/lightstep/otel-launcher-go/lightstep/sdk/metric"
+
+import (
+	"github.com/lightstep/otel-launcher-go/lightstep/sdk/metric/view"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+// config contains configuration options for a MeterProvider.
+type config struct {
+	// res is the resource for this MeterProvider.
+	res *resource.Resource
+
+	// readers is a slice of Reader instances corresponding with views.
+	// the i'th reader uses the i'th entry in views.
+	readers []Reader
+
+	// views is a slice of *Views instances corresponding with readers.
+	// the i'th views applies to the i'th reader.
+	views []*view.Views
```

Comment:
It's not simply to avoid coupling. There is practically no coupling because the SDK is receiving inputs from the instruments and view behavior takes place before the reader begins doing anything. The true coupling happens inside the viewstate in the SnapshotAndProcess()/Collect() methods (as seen in #173).